app/templates/macros/toolkit_forms.html

```diff
@@ -1,300 +1,102 @@
     hint=(question_content.hint or '')|question_references(kwargs.get_question),
-    value=service_data[question_content.id],
-    id=question_content.id,
-    options=question_content.options,
-    type='checkbox_tree',
-    number_of_items=question_content.number_of_items,
-    question_number=kwargs.question_number,
-    hidden=question_content.hidden,
-    error=errors.get(question_content.id)['message']|question_references(kwargs.get_question)
-  %}
-    {% include "toolkit/forms/checkbox-tree.html" %}
-  {% endwith %}
+    question=question_content.question|question_references(kwargs.get_question),
+    **kwargs
+  ) }}
 {%- endmacro %}
 
 {% macro upload(question_content, service_data, errors) -%}
-  {%
-    with
-    question=question_content.question|question_references(kwargs.get_question),
-    question_advice=question_content.question_advice,
+  {{ upstream_forms.upload(question_content, service_data, errors,
+    error=errors.get(question_content.id)['message']|question_references(kwargs.get_question),
     hint=(question_content.hint or '')|question_references(kwargs.get_question),
-    optional=question_content.optional,
-    name=question_content.id,
+    question=question_content.question|question_references(kwargs.get_question),
     value="Document uploaded {}".format(
       service_data[question_content.id]|parse_document_upload_time|datetimeformat
     ) if service_data[question_content.id] else service_data[question_content.id],
-    error=errors.get(question_content.id)['message']|question_references(kwargs.get_question),
-    question_number=kwargs.question_number,
-    message=kwargs.message
-  %}
-    {% include "toolkit/forms/upload.html" %}
-  {% endwith %}
+    **kwargs
+  ) }}
 {%- endmacro %}
 
 {% macro pricing(question_content, service_data, errors) -%}
-  {%
-    with
-    name=question_content.id,
-    question=question_content.question,
-    question_advice=question_content.question_advice,
-    fields=question_content.fields,
-    optional_fields=question_content.optional_fields,
-    price=service_data.get(question_content.fields.price),
-    minimum_price=service_data.get(question_content.fields.minimum_price),
-    maximum_price=service_data.get(question_content.fields.maximum_price),
-    price_unit=service_data.get(question_content.fields.price_unit),
-    price_interval=service_data.get(question_content.fields.price_interval),
-    hours_for_price=service_data.get(question_content.fields.hours_for_price),
-    hint=(question_content.hint or ''),
-    optional=question_content.optional,
-    id=question_content.id,
+  {{ upstream_forms.pricing(question_content, service_data, errors,
     error=errors.get(question_content.id)['message']|question_references(kwargs.get_question),
-    question_number=kwargs.question_number,
-    message=kwargs.message
-  %}
-    {% include "toolkit/forms/pricing.html" %}
-  {% endwith %}
+    hint=(question_content.hint or '')|question_references(kwargs.get_question),
```

Comment:
`hint` and `question` didn't previously have the `|question_references` - why have you added it for this pricing macro, but not for multiquestion and boolean_list?  Do we know we need it here, or it just feels like we might?

Reply:
Ah yes I did add that to some of the parameters for symmetry, assuming they were omissions. I can't think how including this feature generally across all similar fields could backfire too much, and it's the "least surprising" thing for the macros to do for a macro user.